procedures/crates.md

```diff
@@ -0,0 +1,141 @@
+compiler would benefit from a name that is prefixed with `rustc_`. This is an indicator of how
+stable the crate may be to prospective users. Other crates, which are more general-purpose, will
+have names that are disentangled from the compiler.
+
+### Are there any limitations on the review policy for out-of-tree crates?
+Generally, the working groups and team members that are primarily free to maintain the crate using
+whatever practices are best suited to their group, however, there are some limitations so that there
+is some uniformity across the compiler and out-of-tree crates:
+
+- Every compiler team member must be able to review and approve PRs.
+- Where possible, only active participants in the crate (or related working group) need be on the
+  highfive rotation for the crate.
+- Working group members (or regular contributors to the crate) who cannot approve PRs on rustc can
+  approve PRs to the crate.
+- Major pull requests should have multiple reviewers.
+
+### What is required of an out-of-tree crate?
+It is required that out-of-tree crates must:
+
+- Be dual-licensed with Apache 2.0 and MIT (as the compiler is) or
+  [any other license][licenses] accepted by tidy, unless otherwise appropriate (ports of code
+  from other project, etc).
+- Abide by Rust's code of conduct.
+- Specify that the crate is maintained by the Rust compiler team and any appropriate working groups.
+    - In particular, this should detail the expected level of maintenance and stability for any
+      prospective users.
+    - This should also link to the working group details in this repository.
+- Be added to the list at the bottom of this page.
+- Follow semantic versioning.
+- Use `@bors` and `@rust-highfive`.
+
+### Is there a requirement for community infrastructure for an out-of-tree crate?
+There is no requirement that community infrastructure (such as Zulip servers/streams) be created for
+out-of-tree crates. This may be desirable if an out-of-tree crate gains a large community of
+contributors and users, but otherwise, the working group or compiler team streams should be used
+initially.
+
+Linkifiers for auto-linking to issues and PRs on the primary Rust Zulip server can be added on
+request.
+
+### Are there any recommendations for working with out-of-tree crates?
+Recommendations for working with out-of-tree crates will be documented in the rustc-guide, see
+[rust-lang/rustc-guide#285][guide_issue] for progress.
+
+[licenses]: https://github.com/rust-lang/rust/blob/master/src/tools/tidy/src/deps.rs#L10-L19
+[guide_issue]: https://github.com/rust-lang/rustc-guide/issues/285
+
+---
+
+In summary, the process for establishing an out-of-tree crate is as follows:
+
+1. Where appropriate, discuss and confirm the need within the working group for the out-of-tree
+   crate.
+2. Consult with the compiler team to confirm the need for the out-of-tree crate and kick-off the
```

Comment:
I think we should be more specific about this. I would suggest:

To make the approval official, open a PR modifying this document and use rfcbot fcp. 

However, we'll have to make sure we are integrated with rfcbot fcp on this repository (which should be easy enough to do, I think).

Reply:
Added.